### ISSUE TITLE
[Product Configuration Editor] Validate Product: create error markers in "Problems" view

### ISSUE THREAD

**Issue by anb0s** [SSEHUB/EASyProducer#63]
"Validate Product" should create error markers instead of showing modal dialog only.

**Comment by Elscha** [comment]
Dear Andre,

Currently, we also display the error message inside the graphical editor at the failing (top level) variable. So far it is _not_ displayed at the nested elements, but it should not be a great problem to break it down to them. Do you have a suggestion to improve this view?
![errormsg](https://cloud.githubusercontent.com/assets/6489033/14248674/26711dc0-fa76-11e5-9854-0d41b20d28b8.png)

CC: @eichelbe

**Comment by anb0s** [comment]
Yes, i know about the errors in the table view. But first they are shown in
a modal dialog and here it's difficult to find all different errors coming
from different constraints. The same issue is with one line in table view.

Ideas:
- it would be good to have them shown at nested elements if possible -->
  easier to find (should i create other request for this ?)
- in eclipse the developers do not want to see a modal dialog with lot of
  information (text) inside,  so they want to see distinct warnings or errors
  in error view for easy navigation, sorting etc. I think it's some more work
  to do here, because of handling for error markers creation, deletion and
  navigation...

Thanks!

2016-04-04 15:03 GMT+02:00 Sascha El-Sharkawy notifications@github.com:

> Dear Andre,
> 
> Currently, we also display the error message inside the graphical editor
> at the failing (top level) variable. So far it is _not_ displayed at the
> nested elements, but it should not be a great problem to break it down to
> them. Do you have a suggestion to improve this view?
> [image: errormsg]
> https://cloud.githubusercontent.com/assets/6489033/14248674/26711dc0-fa76-11e5-9854-0d41b20d28b8.png
> 
> CC: @eichelbe https://github.com/eichelbe
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> https://github.com/SSEHUB/EASyProducer/issues/63#issuecomment-205287749

## 

Regards
Andre (anb0s)
eMail: anb0s@anbos.de

**Comment by Elscha** [comment]
I added support for marking nested elements as shown in the picture:
![error](https://cloud.githubusercontent.com/assets/6489033/14273988/e64d58a0-fb0e-11e5-9b52-ffd05541f2f5.png)
The internally used vistor needs a test case.

@rsizonenko: Could you maybe:
- Simplyfy the dialog
- Think about the creation of error markers

**Comment by anb0s** [comment]
Thanks, i've tested the nightly build today:
- the creation of errros at nested element is working!
  ![easyproducer_show_nested_errors_01](https://cloud.githubusercontent.com/assets/95811/14378533/bcfd9da2-fd75-11e5-8502-d9779f57d139.png)
- but after resolve the errors only top element is updated (error removed) and all nested have errors, but should not
  ![easyproducer_show_nested_errors_02](https://cloud.githubusercontent.com/assets/95811/14378557/e146585c-fd75-11e5-9ab0-ae27ba698f83.png)

**Comment by anb0s** [comment]
I want to try the version 1.2.0 now. Should i expect this issue fixed or is there a workaround?
Thanks!
